dulwich/graph.py

```diff
@@ -74,17 +136,24 @@ def _has_candidates(wlst, cstates):
                 # do not add it to the working list again
                 if ((pflags & cflags) == cflags):
                     continue
+                pdt = lookup_stamp(pcmt)
+                if pdt < min_stamp:
+                    continue
                 cstates[pcmt] = pflags | cflags
-                wlst.append(pcmt)
+                wlst.add((pdt, pcmt))
 
     # walk final candidates removing any superseded by _DNC by later lower _LCAs
+    # remove any duplicates and sort it so that earliest is first
     results = []
-    for cmt in cands:
-        if not ((cstates[cmt] & _DNC) == _DNC):
-            results.append(cmt)
-    return results
+    for dt, cmt in cands:
+        if not ((cstates[cmt] & _DNC) == _DNC) and not (dt, cmt) in results:
+            results.append((dt, cmt))
+    results.sort(key=lambda x: x[0])
+    lcas = [cmt for dt, cmt in results]
+    return lcas
 
 
+# actual git sorts these based on commit times
```

Comment:
not sure what this comment is trying to do? IIRC git supports both topological sort and sort based on commit times

Reply:
That comment is meant to explain the change to use commit time for this routine.
Git'ss  lca routine the commit time is now used almost exclusively or it is approximated by "adjusted" commit times, but that requires a full tree walk.